addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -131,6 +131,9 @@ def download_efaktur(self):
             if record.partner_id.commercial_partner_id.l10n_id_pkp and not record.l10n_id_tax_number:
                 raise ValidationError(_('Connect %(move_number)s with E-faktur to download this report', move_number=record.name))
 
+            if not record.l10n_id_tax_number:
+                raise UserError(_('Could not download E-faktur without a Tax Number'))
```

Comment:
We were already doing this check just above, but only if `record.partner_id.commercial_partner_id.l10n_id_pkp`

Any reason? :thinking: 

Also, that message is giving which move is in error, which is useful since it's an action in the list view on multiple records...

Reply:
1.
> We were already doing this check just above, but only if `record.partner_id.commercial_partner_id.l10n_id_pkp`
> 
> Any reason? 🤔

That validation error is only raised if the customer belongs to Indonesia. As we can see that 
https://github.com/odoo/odoo/blob/5ba4247574313efd848a9b959e6094cc5216052a/addons/l10n_id_efaktur/models/res_partner.py#L34

2.
> Also, that message is giving which move is in error, which is useful since it's an action in the list view on multiple records...

I have also changed a message for multiple records, as per your suggestion

Reply:
> That validation error is only raised if the customer belongs to Indonesia

But why don't we just remove the condition on `l10n_id_pkp` if it needs to always raise?

Reply:
> > That validation error is only raised if the customer belongs to Indonesia
> 
> But why don't we just remove the condition on `l10n_id_pkp` if it needs to always raise?

The code has been updated now validation error is always raised if downloading an 'Efaktur' without 'l10n_id_tax_number'